File: Cpp/Lista 1/001/001.cpp
//Implementacao da class account

#include <iostream> //Pre-processamento de entrada e saida de dados
using std::cout;
using std::cin;
using std::endl;

//biblioteca matematica
#include <cmath>
//criacao da classe Accoutn

class Accout {
public:
    void setValorDaConta(float reais) {
        saldo = reais;
    }

    void getValorDaConta() {
        cout << "O saldo da sua conta configura: " << saldo <<"R$ " << endl;
    }

    void setDeposito(float valor) {
        saldo = saldo + valor;
        getValorDaConta();
    }

    void setSaque(float valor) {
        if(saldo < valor){
            saldo = saldo - 5;
            getValorDaConta();
            return;
        }

        saldo = saldo - valor;

        getValorDaConta();
    }

    void jurosAno() {
        int i = 1;
        float temp = saldo;

        while(temp < (2 * saldo)) {
            temp = saldo * pow((1 + 0.005), i);
            i = i + 1;
        }

        cout << "Voce ira ter o dobro do dinheiro em " << (i / 12) << " anos! " << endl;
    }


private:
     float saldo;
};

int main() {
    int op;
    float quantia, valor;
    Accout User1;

    cout << "Informe o valor inicial da sua conta: ";
    cin >> quantia;

    system("cls");

    User1.setValorDaConta(quantia);
    User1.getValorDaConta();

    cout << "\nInforme a operacao a ser realizada: \n1 - Para Depositar \n2 - Sacar \nopcao: ";
    cin >> op;

    cout << endl;

    switch(op) {
    case 1:
        cout << "Informe o valor de deposito: ";
        cin >> valor;

        User1.setDeposito(valor);
        break;

    case 2:
        cout << "Informe o valor a ser retirado: ";
            cin >> valor;

            User1.setSaque(valor);
            break;
    }

    return 0;
}

File: Python/Lista 1/001/001.py
#Biblioteca de comandos para o sistema operacional
import os

#Conta
class Conta:

    #Função de inicialização
    def __init__(self, saldo, valor):
        self.saldo = saldo
        self.valor = valor

    #Função Deposito
    def setDeposito(self, valor):
        self.valor = valor
        print(self.valor)
        self.saldo += self.valor

    #Função Saque
    def setSaque(self, valor):
        self.valor = valor

        if(self.valor > self.saldo):
            self.saldo -= 5
            print("Saldo insuficiene, multa de R$5. Seu saldo é de R${}\n".format(self.saldo))

        else:
            self.saldo -= self.valor

    #Mostrar Saldo
    def mostrarSaldo(self):
        print("O seu saldo é de R${}.\n".format(self.saldo))

#Variavel de controle
opcao = None

#Inserindo valor do saldo
conta1 = Conta(float(input("Informe o valor do saldo: ")), 0)

#Menu
while (opcao != 0):
    opcao = int(input("1 - Depositar \n2 - Sacar   \n3 - Ver Saldo  \n0 - Sair  \nInforme sua opção: "))

    if(opcao == 1):
        os.system("cls")
        valor = float(input("Informe o valor do deposito: "))
        conta1.setDeposito(valor)
    
    elif(opcao == 2):
        os.system("cls")
        valor = float(input("Informe o valor do saque: "))
        conta1.setSaque(valor)
    
    elif(opcao == 3):
        os.system("cls")
        conta1.mostrarSaldo()